scss/_patterns_table-expanding.scss

```diff
@@ -39,7 +39,7 @@
         grow: 1;
       }
 
-      &.p-table-expanding__panel {
+      &.p-table__expanding-panel {
```

Comment:
I get confused with BEM in these sorts of situations - looking at [the BEM faqs](http://getbem.com/faq/#block-modifier-affects-elements), it's obviously OK to style child elements within a modifier block, but I haven't found guidance on when an element is only ever intended to be a child of a modifier. 

In this case, `.p-table__expanding-panel` doesn't feature anywhere other than as a child of `.p-table--expanding`, but its naming might lead you to think that you could use `.p-table__expanding-panel` in `.p-table--sortable`.

I would have thought that would be reason enough for it to be a pattern distinct from the one being modified, but that's what this PR is moving away from, so I'm not sure - is there any call for a `p-table__panel`/`p-table__full-width-cell`/etc in other `p-table` patterns?

Reply:
Standard BEM has a simple solution - just chain all the classes you need, so `.p-table.p-table--sortable.p-table--expanding` and inside `.p-table__expanding-panel`. In our modified convention I can't think of a satisfying answer.

Reply:
Yes, this is a weird case. I didn't want to make it `p-table--expanding__panel`, (a child under a modifier) as it seems like a hack and something we don't have anywhere in Vanilla.

I considered `p-table__panel`, but it would only make sense in case if we ever need any kind of panel in other tables. More so, this particular one is used in very specific context, on a cell that expands into whole new row. I doubt we will do anything similar in other tables.
When we implement expanding rows in modular table I believe we should approach it differently.
So, to keep the semantic meaning that this is explicilty a panel that is expanded, I made it `__expanding-panel`.

Current expading table implementation is very specific, it drops table layout for flexbox which has a lots of weird consequences. In longer term, when expanding functionality is impemented as part of modular table, I think we should deprecate and drop current `p-table--expanding`.